src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.exceptions.EmptyDetailsOfTaskError;
+import duke.exceptions.WrongMarkException;
+import duke.tasks.TaskList;
+import duke.tasks.Task;
+import duke.exceptions.UnknownCommandException;
+
+/**
+ * Describes the actions taken by the bot when commands is given
+ */
+public class Parser {
+    /**
+     * Describes the action taken
+     * 
+     * @param command  User command
+     * @param taskList Takes in a taskList to add/delete/change details
+     *                 in the taskList from done to not done and vice
+     *                 versa
+     * @param helper   Takes in the bot interaction with the user
+     * @param storage  Takes in the storage to store the taskList
+     *                 of the particular user after the user wants to end
+     *                 the program of the bot
+     * @return a boolean that allow the main application to end the program
+     *         when a particular command "bye" is given from the user
+     * @throws WrongMarkException      if asked to mark a marked file or unmark a
+     *                                 file
+     *                                 that is not marked
+     * @throws UnknownCommandException if asked commands that the bot do not
+     *                                 understand
+     */
+    public static boolean isExitOrContinue(String command,
+            TaskList tasks,
+            UI helper,
+            Storage storage)
+            throws WrongMarkException,
+            UnknownCommandException {
+        String[] splittedCommand = command.split(" ");
+        String commandType = splittedCommand[0];
+        switch (commandType) {
+            case "bye":
```

Comment:
Should each "case" be at the same indentation level as "switch" based on coding standards?